core/domain/story_services.py

```diff
@@ -535,3 +536,83 @@ def save_story_summary(story_summary):
     )
 
     story_summary_model.put()
+
+
+def get_completed_node_ids(user_id, story_id):
+    """Returns the ids of the nodes completed in the story.
+
+    Args:
+        user_id: str. ID of the given user.
+        story_id: str. ID of the story.
+
+    Returns:
+        list(str). List of the node ids completed in  story.
+    """
+    progress_model = user_models.StoryProgressModel.get(user_id, story_id)
+
+    return progress_model.completed_nodes if progress_model else []
+
+
+def get_node_ids_completed_in_stories(user_id, story_ids):
+    """Returns the ids of the nodes completed in each of the stories.
+
+    Args:
+        user_id: str. ID of the given user.
+        story_ids: list(str). IDs of the stories.
+
+    Returns:
+        list(list(str)). List of the node ids completed in each
```

Comment:
It would be better to have it as a dict keyed by story_id.

Reply:
That works too.

Reply:
Changed.